src/components/Pressable.tsx

```diff
@@ -0,0 +1,281 @@
+   */
+  testOnly_pressed?: null | boolean;
+
+  /**
+   * Either view styles or a function that receives a boolean reflecting whether
+   * the component is currently pressed and returns view styles.
+   */
+  style?:
+    | StyleProp<ViewStyle>
+    | ((state: PressableStateCallbackType) => StyleProp<ViewStyle>);
+
+  /**
+   * Duration (in milliseconds) to wait after press down before calling onPressIn.
+   */
+  unstable_pressDelay?: number;
+}
+
+function touchWithinBounds(touch: TouchData, bounds: Insets): boolean {
+  const leftbound = bounds.left ? bounds.left < touch.absoluteX : true;
+  const rightbound = bounds.right ? bounds.right > touch.absoluteX : true;
+  const bottombound = bounds.bottom ? bounds.bottom < touch.absoluteY : true;
+  const topbound = bounds.top ? bounds.top > touch.absoluteY : true;
+
+  return leftbound && rightbound && topbound && bottombound;
+}
+
+export default function Pressable(props: PressableProps) {
+  const previousTouchData = useRef<TouchData[] | null>(null);
+
+  const pressRetentionOffset: Insets | null | undefined =
+    typeof props.pressRetentionOffset === 'number'
+      ? {
+          top: props.pressRetentionOffset,
+          left: props.pressRetentionOffset,
+          bottom: props.pressRetentionOffset,
+          right: props.pressRetentionOffset,
+        }
+      : props.pressRetentionOffset;
+
+  const touch = Gesture.Manual()
+    .onTouchesDown((event) => {
+      // note: hitslop checking support is built in
+      props.onPressIn?.(event);
+      previousTouchData.current = event.allTouches;
+    })
+    .onTouchesUp((event) => {
+      // doesn't call onPressOut untill the last pointer leaves, while within bounds
+      if (event.allTouches.length > 1) {
+        return;
+      }
+
+      if (!pressRetentionOffset) {
+        props.onPressOut?.(event);
+        return;
+      }
+
+      if (
+        previousTouchData.current?.find((touch) =>
+          touchWithinBounds(touch, pressRetentionOffset)
+        )
+      ) {
+        props.onPressOut?.(event);
+      }
+    });
+
+  const press = Gesture.LongPress().onEnd((event, success) => {
+    if (success) {
+      props.onLongPress?.(event);
+    }
+  });
+
+  const hover = Gesture.Hover()
+    .onBegin((event) => {
+      setTimeout(
+        () => props.onHoverIn?.(event),
+        props.delayHoverIn ?? DEFAULT_HOVER_DELAY
+      );
+    })
+    .onEnd((event) => {
+      setTimeout(
+        () => props.onHoverOut?.(event),
+        props.delayHoverOut ?? DEFAULT_HOVER_DELAY
+      );
+    });
+
+  press.minDuration(props.delayLongPress ?? DEFAULT_LONG_PRESS_DURATION);
+
+  // onBlur and onFocus don't exist in the docs
+
+  touch.hitSlop(props.hitSlop);
+  press.hitSlop(props.hitSlop);
+  hover.hitSlop(props.hitSlop);
+
+  // add props.pressRetentionOffset, according to docs, they're relative to pressable, not hitSlop
+
+  touch.enabled(!(props.disabled ?? false));
```

Comment:
```suggestion
  touch.enabled(props.disabled !== true);
```

Reply:
i think `!== false` would work as intended, since we want all `true`, `undefined` and `null` to be truthy as well

Reply:
You're right